controllers/testutil/util.go

```diff
@@ -40,60 +40,60 @@ func ConfigMapsAreEqual(expected v1.ConfigMap, actual v1.ConfigMap) bool {
 	return true
 }
 
-func DeploymentsAreEqual(dp1 appsv1.Deployment, dp2 appsv1.Deployment) bool {
+func DeploymentsAreEqual(expectedDep appsv1.Deployment, actualDep appsv1.Deployment) bool {
 
-	if !reflect.DeepEqual(dp1.ObjectMeta.Labels, dp2.ObjectMeta.Labels) {
+	if !reflect.DeepEqual(expectedDep.ObjectMeta.Labels, actualDep.ObjectMeta.Labels) {
 		notEqualMsg("labels")
 		return false
 	}
 
-	if !reflect.DeepEqual(dp1.Spec.Selector, dp2.Spec.Selector) {
+	if !reflect.DeepEqual(expectedDep.Spec.Selector, actualDep.Spec.Selector) {
 		notEqualMsg("selector")
 		return false
 	}
 
-	if !reflect.DeepEqual(dp1.Spec.Template.ObjectMeta, dp2.Spec.Template.ObjectMeta) {
+	if !reflect.DeepEqual(expectedDep.Spec.Template.ObjectMeta, actualDep.Spec.Template.ObjectMeta) {
 		notEqualMsg("selector")
 		return false
 	}
 
-	if !reflect.DeepEqual(dp1.Spec.Template.Spec.Volumes, dp2.Spec.Template.Spec.Volumes) {
+	if !reflect.DeepEqual(expectedDep.Spec.Template.Spec.Volumes, actualDep.Spec.Template.Spec.Volumes) {
 		notEqualMsg("Volumes")
 		return false
 	}
 
-	if len(dp1.Spec.Template.Spec.Containers) != len(dp2.Spec.Template.Spec.Containers) {
+	if len(expectedDep.Spec.Template.Spec.Containers) != len(actualDep.Spec.Template.Spec.Containers) {
 		notEqualMsg("Containers")
 		return false
 	}
-	for i := range dp1.Spec.Template.Spec.Containers {
-		c1 := dp1.Spec.Template.Spec.Containers[i]
-		c2 := dp2.Spec.Template.Spec.Containers[i]
-		if !reflect.DeepEqual(c1.Env, c2.Env) {
+	for i := range expectedDep.Spec.Template.Spec.Containers {
+		expectedContainer := expectedDep.Spec.Template.Spec.Containers[i]
+		actualContainer := actualDep.Spec.Template.Spec.Containers[i]
+		if !reflect.DeepEqual(expectedContainer.Env, actualContainer.Env) {
```

Comment:
this (essentially) case block feels pretty inelastic, it would be nice if we could define a list of attributes and dynamically iterate through them

Reply:
yeah I agree, the problem is we do not want to compare all fields (some fields will always be different, e.g. `creationtimestamp`, and there will be some opinionated stances on what ought to be compared, which is probably why an equals operation isn't provided for us already by client-go. I'm open to suggestions though.

Reply:
One suggestion I've seen is to hash a resource and store it in an annotation before submitting it to the api server, so you can always retrieve it's "vanilla" state before the cluster adds it's all cluster specific bloat. Then you can retrieve it and compare it with another hashed resource. 

I wasn't a huge fan of this method though.